src/openvic-simulation/scripts/Condition.cpp

```diff
@@ -0,0 +1,659 @@
+		return false;
+	}
+
+	if (value_type == 0) {
+		Logger::error("Condition ", identifier, " accepts no value type!");
+		return false;
+	}
+
+	if (value_type > 0b1111111) {
+		Logger::error("Condition ", identifier, " accepts an invalid value type!");
+		return false;
+	}
+
+	if (scope == 0) {
+		Logger::error("Condition ", identifier, " has no scope specified!");
+		return false;
+	}
+
+	if (scope > (scope_t::COUNTRY | scope_t::PROVINCE | scope_t::POP | scope_t::MAGIC)) {
+		Logger::error("Condition ", identifier, " has no scope specified!");
+		return false;
+	}
+
+	if ((value_identifier_type == identifier_type_t::NONE) && (value_type == value_type_t::IDENTIFIER)) {
+		Logger::error("Condition ", identifier, " has no identifier type!");
+		return false;
+	}
+
+	if ((value_identifier_type != identifier_type_t::NONE) && (value_type != value_type_t::IDENTIFIER)) {
+		Logger::warning("Condition ", identifier, " specified an identifier type, but doesn't have an identifier!");
+	}
+
+	return conditions.add_item({ 
+		identifier,
+		value_type,
+		scope,
+		scope_change,
+		key_identifier_type,
+		value_identifier_type
+	});
+}
+
+bool ConditionManager::setup_conditions(GameManager const& game_manager) {
+	bool ret = true;
+
+	using enum value_type_t;
+	using enum scope_t;
+	using enum identifier_type_t;
+
+	/* Magic Scopes */
+	ret &= add_condition("THIS", GROUP, COUNTRY | PROVINCE | POP, MAGIC);
+	ret &= add_condition("FROM", GROUP, COUNTRY | PROVINCE | POP, MAGIC);
+
+	/* Mixed Scopes */
+	ret &= add_condition("any_core", GROUP, COUNTRY | PROVINCE, PROVINCE);
+	ret &= add_condition("any_pop", GROUP, COUNTRY | PROVINCE, POP);
+	ret &= add_condition("cultural_union", GROUP, COUNTRY | POP, COUNTRY);
+
+	/* Trigger Country Scopes */
+	ret &= add_condition("all_core", GROUP, COUNTRY, PROVINCE);
+	ret &= add_condition("any_greater_power", GROUP, COUNTRY, COUNTRY);
+	ret &= add_condition("any_neighbor_country", GROUP, COUNTRY, COUNTRY);
+	ret &= add_condition("any_owned_province", GROUP, COUNTRY, PROVINCE);
+	ret &= add_condition("any_sphere_member", GROUP, COUNTRY, COUNTRY);
+	ret &= add_condition("any_state", GROUP, COUNTRY, STATE);
+	ret &= add_condition("any_substate", GROUP, COUNTRY, COUNTRY);
+	ret &= add_condition("capital_scope", GROUP, COUNTRY, PROVINCE);
+	ret &= add_condition("overlord", GROUP, COUNTRY, COUNTRY);
+	ret &= add_condition("sphere_owner", GROUP, COUNTRY, COUNTRY);
+	ret &= add_condition("war_countries", GROUP, COUNTRY, COUNTRY);
+
+	/* Trigger Province Scopes */
+	ret &= add_condition("any_neighbor_province", GROUP, PROVINCE, PROVINCE);
+	ret &= add_condition("controller", GROUP, PROVINCE, COUNTRY);
+	ret &= add_condition("owner", GROUP, PROVINCE, COUNTRY);
+	ret &= add_condition("state_scope", GROUP, PROVINCE);
+
+	/* Trigger Pop Scopes */
+	ret &= add_condition("location", GROUP, POP, PROVINCE);
+	ret &= add_condition("country", GROUP, POP, COUNTRY);
+
+	/* Magic Conditions */
```

Comment:
Avoid the term magic in code. It can only create confusion.
AND & OR are collections of conditions.
NOT is basically an inverse OR and can thus be seen as a collection.